cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/ConfiguratorFactory.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.internal.app.deploy;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.app.deploy.Configurator;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.id.Id;
+import io.cdap.cdap.internal.app.runtime.artifact.PluginFinder;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.filesystem.Location;
+
+/**
+ * ConfiguratorFactory which returns either a RemoteConfigurator or InMemoryConfigurator
+ */
+public class ConfiguratorFactory {
+
+  private final boolean isRemote;
```

Comment:
Where does this come from? From cConf? If that's the case, inject cConf to the constructor and get the config from there.

Reply:
This has been refactored to use a provider which pulls from the cConf like you suggested